test/appender/test_appender.cpp

```diff
@@ -169,6 +169,38 @@ TEST_CASE("Test AppendRow", "[appender]") {
 	REQUIRE(CHECK_COLUMN(result, 2, {Value::TIMESTAMP(1992, 1, 1, 1, 1, 1, 0)}));
 }
 
+TEST_CASE("Test default value appender", "[appender]") {
+	duckdb::unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE integers(i iNTEGER, j INTEGER DEFAULT 5)"));
```

Comment:
* Can we add a test for `DEFAULT nextval('seq')`?
* Can we add a test for `DEFAULT random()`?
* Can we add a test for `DEFAULT now()`?
* Can we add a default value that triggers an error, e.g.:

```sql
create table integers(i integer default 'hello'::int);
insert into integers default values;
-- Conversion Error: Could not convert string 'hello' to INT32
```

* Can we add a test for the "default default", i.e. returning `NULL` when no value is specified

Reply:
I disabled the "default default", only if the column has a default expression is the use of AppendDefault allowed
It throws an exception otherwise (added a test for that behavior as well)

Reply:
That’s supported in SQL normally, I think it makes sense for this to be equivalent to the SQL DEFAULT expression so I would vote for allowing this